### PR TITLE
Lighten AppCore dependency, remove double-define

### LINKED ISSUE #15: kLastUsedTimeKey is defined twice
To keep track of last time the app was used, the app delegate sets `kLastUsedTimeKey` on termination/moving to background. APCTabBarViewController does the same – with the same key that it defines itself – when PIN entry succeeds.

It seems the tab controllers implementation can be removed entirely. If not it should probably be moved to the delegate method that the app delegate implements and the define should be removed.

### LINKED ISSUE #16: #import "APCAppCore.h" where it's not necessary
Both `APCTabBarViewController` and `APCPasscodeViewController` import all AppCore headers, making it impossible to use them outside of the whole bundle.

### PR DESCRIPTION
**Pull request by p2**
Fixes ResearchKit/AppCore#15 and fixes ResearchKit/AppCore#16

### CODE REVIEW

File: APCAppCore/APCAppCore/Startup/APCTabBarViewController.m [hunk 1]
```diff
@@ -67,8 +66,6 @@ - (void)passcodeViewControllerDidSucceed:(APCPasscodeViewController *)viewContro
     self.isPasscodeShowing = NO;
     self.showPasscodeScreen = NO;
     [viewController dismissViewControllerAnimated:YES completion:nil];
-    [[NSUserDefaults standardUserDefaults] setObject: [NSDate date] forKey:kLastUsedTimeKey];
```
Comment (peculiar) [inline APCAppCore/APCAppCore/Startup/APCTabBarViewController.m:70]: Capturing the timestamp of when the user was authenticate by the app is an approximation for the _Last Used_ timestamp in the event the user deletes the app from the _Recently Used_ list. As such, these two lines cannot be deleted unless there's another mechanism for capturing the _Last Used_ timestamp.

Comment (p2) [inline APCAppCore/APCAppCore/Startup/APCTabBarViewController.m:70]: These 2 lines can be added to AppDelegate's `passcodeViewControllerDidSucceed` implementation.

**Comment by jwe-apple** [comment]
@p2 It would be easier to deal with this if you kept different changes separate. Naming changes, for instance are very easy to accept. Changes to APCAppDelegate behavior are harder to accept because we need to check for overrides in the app projects. Could you split this PR please?

**Comment by p2** [comment]
It seems the PR gets updated whenever I push something, that was not my intention, sorry for that. How can I fix this?

**Comment by jwe-apple** [comment]
Put your changes in separate branches. We could close this PR, and you should create a separate branch for each change and PR them separately. If you need to use them all together in the meantime, then merge them all into your working branch in your fork.

**Comment by p2** [comment]
Alright, let me do that. Thanks.

**Pull request closed without merging**